index.js

```diff
@@ -52,12 +56,21 @@ function keyPair (storage, derive) {
     function createNew () {
       derive(null, (err, res) => {
         if (err) return cb(err)
-        writeName(res.name, err => {
+        keyStorage.write(0, res.publicKey, err => {
           if (err) return cb(err)
-          keyStorage.write(0, res.publicKey, err => {
-            if (err) return cb(err)
-            return cb(null, res)
-          })
+          if (res.name) {
+            writeName(res.name, err => {
+              if (err) return cb(err)
+              return cb(null, res)
+            })
```

Comment:
moving the callback here to a done func and using it below as well found make this a bit less verbosse